From f3bfef392845b4279f053db0d6050f0f32c26b23 Mon Sep 17 00:00:00 2001
From: Eric Ernst <eric.ernst@intel.com>
Date: Tue, 13 Oct 2015 09:54:44 -0700
Subject: [PATCH] x86: configs: add debug_diffconfig

This diffconfig is intended to be used at build time in conjunction
with CHT's x86_64_defconfig to generate a config for debug builds,
and should replace the use of x86_64_debug_defconfig.

Change-Id: Iafc38ec945cf6575bde5d80e6b746d7ac73cdc22
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5220
Signed-off-by: Eric Ernst <eric.ernst@intel.com>
---
 arch/x86/configs/debug_diffconfig | 41 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 arch/x86/configs/debug_diffconfig

diff --git a/arch/x86/configs/debug_diffconfig b/arch/x86/configs/debug_diffconfig
new file mode 100644
index 0000000..f578a87
--- /dev/null
+++ b/arch/x86/configs/debug_diffconfig
@@ -0,0 +1,41 @@
+#Debug configuration options
+CONFIG_DEBUG_SG=y
+CONFIG_DEBUG_LIST=y
+CONFIG_RCU_CPU_STALL_INFO=y
+CONFIG_DEBUG_MUTEXES=y
+CONFIG_DEBUG_SPINLOCK=y
+CONFIG_LOCKUP_DETECTOR=y
+CONFIG_HARDLOCKUP_DETECTOR_NMI=y
+CONFIG_HARDLOCKUP_DETECTOR=y
+CONFIG_BOOTPARAM_HARDLOCKUP_PANIC=y
+CONFIG_BOOTPARAM_HARDLOCKUP_PANIC_VALUE=1
+CONFIG_BOOTPARAM_SOFTLOCKUP_PANIC=y
+CONFIG_BOOTPARAM_SOFTLOCKUP_PANIC_VALUE=1
+CONFIG_DETECT_HUNG_TASK=y
+CONFIG_DEFAULT_HUNG_TASK_TIMEOUT=120
+CONFIG_BOOTPARAM_HUNG_TASK_PANIC=y
+CONFIG_BOOTPARAM_HUNG_TASK_PANIC_VALUE=1
+CONFIG_PANIC_ON_OOPS=y
+CONFIG_PANIC_ON_OOPS_VALUE=1
+CONFIG_KPROBES=y
+CONFIG_KPROBES_ON_FTRACE=y
+CONFIG_KRETPROBES=y
+CONFIG_KPROBE_EVENT=y
+CONFIG_UPROBES=y
+CONFIG_UPROBE_EVENT=y
+CONFIG_PROBE_EVENTS=y
+CONFIG_DEBUG_KMEMLEAK=y
+CONFIG_DEBUG_KMEMLEAK_EARLY_LOG_SIZE=1000
+# CONFIG_DEBUG_KMEMLEAK_TEST is not set
+CONFIG_DEBUG_KMEMLEAK_DEFAULT_OFF=y
+# CONFIG_KPROBES_SANITY_TEST is not set
+# CONFIG_NET_TCPPROBE is not set
+# CONFIG_X86_DECODER_SELFTEST is not set
+CONFIG_PERCPU_RWSEM=y
+CONFIG_X86_MSR=y
+CONFIG_DEVMEM=y
+# CONFIG_STRICT_DEVMEM is not set
+CONFIG_CPU_FREQ_STAT_DETAILS=y
+CONFIG_LBR_DUMP_ON_EXCEPTION=y
+CONFIG_DEBUG_PANIC_ON_BAD_PAGE=y
+CONFIG_INTEL_TASKS_DUMP=y
-- 
1.9.1

